core/src/com/unciv/ui/utils/ImageGetter.kt

```diff
@@ -67,6 +67,50 @@ object ImageGetter {
         }
     }
 
+
+    /**
+     * Colors a multilayer image and returns it as a list of layers (Image).
+     *
+     * @param   baseFileName    The filename of the base image.
+     *                              For example: TileSets/FantasyHex/Units/Warrior
+     *
+     * @param   colors          The list of colors, one per layer. No coloring is applied to layers
+     *                              whose color is null.
+     *
+     * @return  The list of layers colored. The layers are sorted by alphabetical order and colors
+     *              are applied, one per layer, in the same order. If a color is null, no coloring is
+     *              performed on such layer (it stays as it is). If there are less colors than layers,
+     *              the last layers are not colored. Defaults to an empty list if there is no layer
+     *              corresponding to baseFileName.
+     *
+     * Example:
+     *      getLayeredImageColored("TileSets/FantasyHex/Units/Warrior", null, Color.GOLD, Color.RED)
+     *
+     *      All images in the atlas that starts with "TileSets/FantasyHex/Units/Warrior" are retrieved
+     *      alphabetically:
+     *          - TileSets/FantasyHex/Units/Warrior
+     *          - TileSets/FantasyHex/Units/Warrior-1
+     *          - TileSets/FantasyHex/Units/Warrior-2
+     *
+     *      It returns a list in which first layer has unmodified colors, the second is colored in
+     *      GOLD and the third in RED.
+     */
+    fun getLayeredImageColored(baseFileName: String, vararg colors: Color?): ArrayList<Image> {
+        val layerList = arrayListOf<Image>()
+        val layers = textureRegionDrawables.keys
+                .filter { it.startsWith(baseFileName) }
```

Comment:
Not good - some units could start with the name of other units, especially in mods
Example: Fighter, Fighter Jet
Warrior, Warrior prince
Zombie, Zombie legion

Reply:
I changed the rules for filtering. It is now based on a regex. Here you can find an example https://regex101.com/r/TgintQ/1